great_expectations/core/util.py

```diff
@@ -427,6 +429,13 @@ def convert_to_json_serializable(  # noqa: C901 - complexity 32
     if StructType is not None and isinstance(data, StructType):
         return dict(data.jsonValue())
 
+    if sqlalchemy is not None and isinstance(data, TextClause):
+        # TextClause is converted to str manually
+        return str(data)
+    if sqlalchemy is not None and isinstance(data, Connection):
+        # Connection is a module, which is non-serializable. Return module name instead.
+        return "sqlalchemy.engine.base.Connection"
```

Comment:
Thank you for catching this @anthonyburdi. This is because the `NotImported` class in `optional_imports.py` has a `__bool__`, and this make it different from the pattern followed by `StructType` in like 429 of this current file?

Reply:
Exactly! You might have missed the meeting where we added that to NotImported.